File: app/src/test/java/com/hana053/micropost/presentation/top/TopActivityTest.java
package com.hana053.micropost.presentation.top;

import com.hana053.micropost.testing.RobolectricBaseTest;

import org.junit.Before;
import org.junit.Test;
import org.robolectric.Robolectric;
import org.robolectric.util.ActivityController;

import static org.hamcrest.Matchers.instanceOf;
import static org.junit.Assert.assertThat;

public class TopActivityTest extends RobolectricBaseTest {

    @SuppressWarnings("FieldCanBeLocal")
    private ActivityController<TopActivity> activityController;
    private TopActivity activity;

    @Before
    public void setup() {
        activityController = Robolectric.buildActivity(TopActivity.class);
        activity = activityController.setup().get();
    }

    @Test
    public void shouldBeCreated() {
        assertThat(activity, instanceOf(TopActivity.class));
    }

}

File: app/src/test/java/com/hana053/micropost/pages/top/TopActivityTest.kt
package com.hana053.micropost.pages.top

import com.hana053.micropost.testing.RobolectricBaseTest
import org.hamcrest.Matchers.instanceOf
import org.junit.Assert.assertThat
import org.junit.Before
import org.junit.Test
import org.robolectric.Robolectric
import org.robolectric.util.ActivityController

class TopActivityTest : RobolectricBaseTest() {

    lateinit var activityController: ActivityController<TopActivity>
    lateinit var activity: TopActivity

//    // TODO remove later
//    val httpErrorHandler: HttpErrorHandler = mock(HttpErrorHandler::class.java)
//    val topPresenter: TopPresenter = mock(TopPresenter::class.java)

    @Before
    fun setup() {
//        overrideAppBindings {
//            bind<HttpErrorHandler>(overrides = true) with instance(httpErrorHandler)
//        }
//        putOverridingModule(TopActivity::class.java, Kodein.Module {
//            bind<TopPresenter>(overrides = true) with instance(topPresenter)
//        })
        activityController = Robolectric.buildActivity(TopActivity::class.java)
        activity = activityController.setup().get()

    }

    @Test
    fun shouldBeCreated() {
//        assertThat(httpErrorHandler, `is`(activity.httpErrorHandler))
//        assertThat(topPresenter, `is`(activity.presenter))
//        assertThat(navigator, `is`(activity.navigator))
        assertThat(activity, instanceOf(TopActivity::class.java))
    }

}